fence/README.md

```diff
@@ -0,0 +1,153 @@
+# Network Fencing
+
+## Terminology
+
+| Term     | Definition                                                                            |
+| -------- | ------------------------------------------------------------------------------------- |
+| CO       | Container Orchestration system that communicates with plugins using CSI service RPCs. |
+| SP       | Storage Provider, the vendor of a CSI plugin implementation.                          |
+| DR       | Disaster Recovery.                                                                    |
+| RPC      | [Remote Procedure Call](https://en.wikipedia.org/wiki/Remote_procedure_call).         |
+
+## Objective
+
+Define a standard that will enable storage vendors (SP) to
+perform node level fencing using corresponding CIDR blocks.
+
+### Goals in MVP
+
+The new extension will define a procedure that allows SP to block
+access to any client using corresponding CIDR blocks, and unblock
+as per the requirement.
+
+If multiple COs are accessing a shared resource provided by the SP, and a node
+failure takes place, this procedure will allow the SP to block access from any
+node of CO, and thus preventing further access and corruption of data.
+
+### Non-Goals in MVP
+
+* Application level fencing, i.e blocklisting by maintaining a list of applications
+  that are to be denied access.
+
+## Solution Overview
+
+This specification defines an interface along with the minimum operational and
+packaging recommendations for a storage provider (SP) to implement fencing operation.
+The interface declares the RPCs that a plugin MUST expose.
+
```

Comment:
There is no mentioning if this functionality is expected to be implemented on the CSI Controller (provisioner) or CSI Node. Is the location relevant at all?

Reply:
We can do blocklisting from either provisioner or node as it is just a call to the ceph cluster and doesn't involve volumes.
Since it is Network Fencing, Provisioner will be a good place to do it

Reply:
If you expect that the CSI Controller (provisioner) will do the fencing, make sure to describe that well. The CSI NodePlugin of the CSI-driver will not need to implement any fence operations in that case. A description explaining this design is currently missing.

Reply:
Thanks! Added description for the same